cli/azd/internal/appdetect/pom.go

```diff
@@ -64,53 +72,421 @@ type plugin struct {
 	Version    string `xml:"version"`
 }
 
-func toPom(filePath string) (*pom, error) {
-	bytes, err := os.ReadFile(filePath)
+const (
+	DependencyScopeCompile string = "compile"
+	DependencyScopeTest    string = "test"
+)
+
+func createEffectivePomOrSimulatedEffectivePom(pomPath string) (pom, error) {
+	pom, err := createEffectivePom(pomPath)
+	if err == nil {
+		return pom, nil
+	}
+	return createSimulatedEffectivePom(pomPath)
+}
+
+// Simulated effective pom not strictly equal to effective pom,
+// it just tries best to make sure these item are same to the real effective pom:
+//  1. pom.Dependencies. Only care about the groupId/artifactId/version.
+//  2. pom.Build.Plugins.
+//     2.1. Only care about the groupId/artifactId/version.
+//     2.2. Not include the default maven plugins (name with this patten: "maven-xxx-plugin").
```

Comment:
I'm not 100% sure that current implementation can cover cases. I already added a lot of tests in `pom_test.go`. We can improve this func when we found new uncovered cases.